File: dagger/src/main/java/com/laioffer/lesson/手写dagger/MainActivity.java
package com.laioffer.lesson.手写dagger;

import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.laioffer.R;

import javax.inject.Inject;


/**
 * @description 注入目标
 * @date 2022/10/6 3:19 下午
 */
public class MainActivity extends AppCompatActivity {

  @Inject // 第5个注解
  public Computer student;

  @Inject // 第5个注解
  public Computer student2;

  @Override
  protected void onCreate(Bundle savedInstanceState) { // 方法进栈
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    // 第4个注解@Component生成的类
    DaggerComputerComponent.create().inject(this);

    Toast.makeText(this, "student:" + student.hashCode(), Toast.LENGTH_SHORT).show();

    Toast.makeText(this, "student:" + student2.hashCode(), Toast.LENGTH_SHORT).show();
  }
}

File: fragment/viewpager/src/main/java/com/laioffer/MainActivity.kt
package com.laioffer

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.laioffer.demo1.Demo1Activity
import com.laioffer.fragment.viewpager.R
import com.laioffer.viewpage2.demo1.Vp2Demo1Activity


class MainActivity : AppCompatActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)
    startNewActivity(R.id.viewpager_demo1_btn, Demo1Activity::class.java)
    startNewActivity(R.id.viewpager2_demo1_btn, Vp2Demo1Activity::class.java)
  }

  private fun startNewActivity(buttonRes: Int, clazz: Class<*>) {
    findViewById<View>(buttonRes).setOnClickListener {
      val intent = Intent(this, clazz)
      startActivity(intent)
    }
  }
}